pkg/config/validation/validation.go

```diff
@@ -1306,15 +1306,38 @@ func validateTrafficPolicy(policy *networking.TrafficPolicy) Validation {
 		return Validation{}
 	}
 	if policy.OutlierDetection == nil && policy.ConnectionPool == nil &&
-		policy.LoadBalancer == nil && policy.Tls == nil && policy.PortLevelSettings == nil {
+		policy.LoadBalancer == nil && policy.Tls == nil && policy.PortLevelSettings == nil && policy.Tunnel == nil {
 		return WrapError(fmt.Errorf("traffic policy must have at least one field"))
 	}
 
 	return appendValidation(validateOutlierDetection(policy.OutlierDetection),
 		validateConnectionPool(policy.ConnectionPool),
 		validateLoadBalancer(policy.LoadBalancer),
 		validateTLS(policy.Tls),
-		validatePortTrafficPolicies(policy.PortLevelSettings))
+		validatePortTrafficPolicies(policy.PortLevelSettings),
+		validateTunnelSettings(policy.Tunnel))
+}
+
+func validateTunnelSettings(tunnel *networking.TrafficPolicy_TunnelSettings) (errs error) {
+	if tunnel == nil {
+		return
+	}
+	if tunnel.Protocol == "" {
+		errs = appendErrors(errs, fmt.Errorf("tunnel protocol must be specified"))
+	}
+	if tunnel.Protocol != "connect" && tunnel.Protocol != "post" {
```

Comment:
```suggestion
	if tunnel.Protocol != "CONNECT" && tunnel.Protocol != "POST" {
```

Reply:
Will need an API docs change, but I have only ever seen methods as all caps

Reply:
It checks lowercase, because of this discussion: https://github.com/istio/api/pull/2283#discussion_r840635183. But I also prefer uppercase, so I will open a PR to Istio API.

Reply:
Sorry, you are right - method is upper case. It is pretty confusing to use a method name and call it protocol - in my comment I was assuming we'll define some protocols like http2-connect. 

But see my comment above - examples and typical usage should not include this, POST is for legacy/special cases, and ideally protocol can be negotiated/detected.

For capitalization - just use whatever Envoy is using in their config.